distributed/utils.py

```diff
@@ -386,7 +386,7 @@ def key_split_group(x):
     typ = type(x)
     if typ is tuple:
         return x[0]
-    elif typ is str:
+    elif typ is str or typ is six.text_type:
```

Comment:
This is going to sound a bit extreme, but I'd like to avoid the attribute access here.  I recommend importing `unicode` from `distributed.compatibility` and using that instead